### PR TITLE
replace tbb map with std map

### PR DESCRIPTION
**Pull request by morebtcg**
replace tbb map with std map

### CODE REVIEW

File: libstorage/Table.cpp [hunk 1]
```diff
@@ -323,9 +323,17 @@ ssize_t Entry::refCount()
     return m_data->m_refCount;
 }
 
+std::shared_ptr<Entry::RWMutexScoped> Entry::lock(bool write)
```
Comment (cyjseagull) [inline libstorage/Table.cpp]: This seems useless.

File: libstorage/CachedStorage.cpp [hunk 2]
```diff
@@ -571,7 +554,34 @@ void CachedStorage::init()
         m_ID = boost::lexical_cast<size_t>(numStr);
     }
 
-    startClearThread();
+    if (!disabled())
+    {
+        startClearThread();
```
[the remark this answers is not in the document]
Comment (morebtcg) [reply libstorage/CachedStorage.cpp]: If so, ClearThread will drop all data in m_caches each trigger

File: libstorage/CachedStorage.cpp [hunk 3]
```diff
@@ -674,16 +685,75 @@ std::tuple<std::shared_ptr<Cache::RWScoped>, Cache::Ptr, bool> CachedStorage::to
         ++m_hitTimes;
     }
 
-    return std::make_tuple(cacheLock, cache, locked);
+    return std::make_tuple(cacheLock, cache, true);
```
[the remark this answers is not in the document]
Comment (morebtcg) [reply libstorage/CachedStorage.cpp]: @wheatli

File: libdevcore/ThreadPool.h [hunk 4]
```diff
@@ -53,7 +53,10 @@ class ThreadPool
     void stop()
     {
         _ioService.stop();
-        _workers.join_all();
+        if (!_workers.is_this_thread_in())
```
Comment (bxq2011hust) [inline libdevcore/ThreadPool.h]: this should be impossible. I think this modify is unnecessary😢

File: libstorage/CachedStorage.cpp [hunk 5]
```diff
@@ -571,7 +550,38 @@ void CachedStorage::init()
...
+void CachedStorage::stop()
+{
+    STORAGE_LOG(INFO) << "Stoping flushStorage thread";
+    m_running->store(false);
+
+    m_taskThreadPool->stop();
+
+    if (m_clearThread)
+    {
+        if (m_clearThread->get_id() != std::this_thread::get_id())
```
Comment (bxq2011hust) [inline libstorage/CachedStorage.cpp]: stop() called by m_clearThread should be impossible. I think this modify is unnecessary😢

File: libstorage/CachedStorage.cpp [hunk 6]
```diff
@@ -571,7 +550,38 @@ void CachedStorage::init()
...
+        {
+            m_clearThread->join();
+            m_clearThread.reset();
+        }
+        else
```
Comment (bxq2011hust) [inline libstorage/CachedStorage.cpp]: this else is uncessary

**Pull request merged**